pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,350 @@
+func createMachinePool(opts *options, mprequest *v1.MachinePool) error {
+	// create a new machine pool via ocm
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+	response, err := client.Clusters().Cluster(opts.selectedCluster.ID()).MachinePools().Add().Body(mprequest).Send()
+	if err != nil {
+		return err
+	}
+	opts.selectedClusterMachinePool = *response.Body()
+	return nil
+}
+
+func createMachinePoolInteractivePrompt(opts *options) error {
+	validator := &dedicatedcmdutil.Validator{
+		Localizer:  opts.f.Localizer,
+		Connection: opts.f.Connection,
+	}
+	// TODO add page size and Localizer
+	promptNodeCount := &survey.Input{
+		Message: "Enter the desired machine pool node count",
+		Help:    "The machine pool node count must be greater than or equal to and a multiple of 3",
+	}
+	var nodeCount int
+	err := survey.AskOne(promptNodeCount, &nodeCount, survey.WithValidator(validator.ValidatorForMachinePoolNodes))
+	if err != nil {
+		return err
+	}
+	opts.requestedMachinePoolNodeCount = nodeCount
+	dedicatedMachinePool, err := createMachinePoolRequestForDedicated(nodeCount)
+	if err != nil {
+		return err
+	}
+	err = createMachinePool(opts, dedicatedMachinePool)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// machine pool replica count must be greater than or equal and a multiple of 3
+func validateMachinePoolNodes(opts *options) error {
+	if len(opts.clusterMachinePoolList.Slice()) > 0 {
+		for _, machinePool := range opts.clusterMachinePoolList.Slice() {
+			if validateMachinePoolCount(machinePool.Replicas()) &&
+				checkForValidMachinePoolLabels(*machinePool) &&
+				checkForValidMachinePoolTaints(*machinePool) {
+				opts.selectedClusterMachinePool = *machinePool
+				return nil
+			} else {
+				err := createMachinePoolInteractivePrompt(opts)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func selectAccessPrivateNetworkInteractivePrompt(opts *options) error {
+	options := []string{"Yes", "No"}
+	prompt := &survey.Select{
+		Message: "Do you want to access a private network?",
+		Options: options,
+	}
+	err := survey.AskOne(prompt, &options)
+	if err != nil {
+		return err
+	}
+	if options[0] == "Yes" {
+		opts.accessKafkasViaPrivateNetwork = true
+	} else {
+		opts.accessKafkasViaPrivateNetwork = false
+	}
+
+	return nil
+}
+
+// TODO add machine pool logic
+func registerClusterWithKFM(opts *options) error {
+	clusterIngressDNSName, err := parseDNSURL(opts)
+	if err != nil {
+		return err
+	}
+	kfmPayload := kafkamgmtclient.EnterpriseOsdClusterPayload{
+		AccessKafkasViaPrivateNetwork: opts.accessKafkasViaPrivateNetwork,
+		ClusterId:                     opts.selectedCluster.ID(),
+		ClusterExternalId:             opts.selectedCluster.ExternalID(),
+		ClusterIngressDnsName:         clusterIngressDNSName,
+		KafkaMachinePoolNodeCount:     int32(opts.selectedClusterMachinePool.Replicas()),
```

Comment:
same comment as https://github.com/redhat-developer/app-services-cli/pull/1795/files#r1084128248 if ever the machine pool is autoscaled